### PR TITLE
[WFLY-4357] Update the JDR Subsystem to Include a Type 4 UUID in its Report

### PR DESCRIPTION
**Pull request by Kinlaw**
https://issues.jboss.org/browse/WFLY-4357

### CODE REVIEW

File: jdr/jboss-as-jdr/src/main/java/org/jboss/as/jdr/JdrReport.java [hunk 1]
```diff
@@ -80,4 +101,69 @@ public String getLocation() {
...
+    /**
+     * sets JDR UUID using jdr.properties file
+     * if the JDR does not have a UUID, then generate one for them and store it in the jdr.properties file
+     */
+    private void setJdrUuid() {
+        String jbossConfig = System.getProperty(JBOSS_PROPERTY_DIR);
+        // JDR is being ran from command line
+        if(jbossConfig == null) {
+            // if JBoss standalone directory does not exist then go no further
+            if(!new File(DEFAULT_PROPERTY_DIR).exists()) {
```
Comment (bstansberry) [inline jdr/jboss-as-jdr/src/main/java/org/jboss/as/jdr/JdrReport.java]: Won't work on Windows.

File: jdr/jboss-as-jdr/src/main/java/org/jboss/as/jdr/JdrReport.java [hunk 2]
```diff
@@ -80,4 +101,69 @@ public String getLocation() {
...
+                FileNotFoundException e = new FileNotFoundException("Could not find JDR properties file");
+                e.printStackTrace();
```
Comment (bstansberry) [inline jdr/jboss-as-jdr/src/main/java/org/jboss/as/jdr/JdrReport.java]: EAP exceptions need to be i18n, and should be thrown or logged via a logger, not System.err. The i18n class JdrLogger should be used; e.g. JdrLogger.varNull.

**Comment by Kinlaw** [comment]
@bstansberry issues fixed in 34b6aa9 and ee783de

**Comment by bstansberry** [comment]
As discussed in email, Emmanuel Hugonnet is taking over this branch.

**Pull request closed without merging**